config/optional/diagnostics.nix

```diff
@@ -0,0 +1,40 @@
+{
+  config.vim = {
+    # Enhanced diagnostics for comprehensive gutter information
+    diagnostics = {
+      enable = true;
+      config = {
+        signs = {
+          text = {
+            ERROR = "";
+            WARN = "";
+            INFO = "";
+            HINT = "";
+          };
+          numhl = {
+            ERROR = "DiagnosticSignError";
+            WARN = "DiagnosticSignWarn";
+            INFO = "DiagnosticSignInfo";
+            HINT = "DiagnosticSignHint";
+          };
+          linehl = { };
+        };
+        severity_sort = true;
+        float = {
+          focusable = false;
+          style = "minimal";
+          border = "rounded";
+          source = "always";
+          header = "";
+          prefix = "";
+        };
+        virtual_text = {
+          spacing = 4;
+          source = "if_many";
+          prefix = "●";
```

Comment:
Is there a better symbol we can use here? I'd love it if this was a little clippy!